src/libsyntax/ast.rs

```diff
@@ -906,7 +906,27 @@ pub enum MacStmtStyle {
     NoBraces,
 }
 
-/// Local represents a `let` statement, e.g., `let <pat>:<ty> = <expr>;`.
+/// The interpreter state of a `Local`.
+#[derive(Clone, Copy, PartialEq, Eq, RustcEncodable, RustcDecodable, Debug)]
+pub enum LocalInterpState {
+    /// The local was not uninitialized in any previous evaluation session.
+    Uninitialized,
+    /// The local was set to a value in a previous evaluation session.
+    Set,
+    /// The local was moved during a previous evaluation session.
+    Moved,
+}
```

Comment:
Hm, so to be honest this sounds pretty frightening; I would expect borrowck (and related passes) to be in charge of this, not the interpreter (and for it not to be part of AST, as moves can happen on more than just local variables).

Reply:
It's much less pernicious than you'd think. The thing is, it's nothing to do with borrowck per se, because the sets/moves occurred in a previous compiler session.

Reply:
Well mostly a drive-by comment anyway, it doesn't hurt us all that much :)